rpc/core/mempool.go

```diff
@@ -20,6 +20,9 @@ import (
 // CheckTx nor transaction results.
 // More: https://docs.cometbft.com/main/rpc/#/Tx/broadcast_tx_async
 func (env *Environment) BroadcastTxAsync(_ *rpctypes.Context, tx types.Tx) (*ctypes.ResultBroadcastTx, error) {
+	if env.ConsensusReactor.WaitSync() {
```

Comment:
I think the calls should check against the mempool whether it is accepting transactions.

Reply:
I think that would be better, but then we would need to add a `MempoolReactor` to the environment. Are we ok with that?

Reply:
There are two options here. One is what it's currently implemented: we keep the information that the node is catching up (WaitSync) only in the consensus reactor, and the mempool RPC endpoints query the consensus reactor via the RPC environment. This creates a slight dependency in the mempool endpoints to the consensus reactor. 

The other option is to duplicate the WaitSync information in the mempool reactor and add to the RPC environment a pointer to the mempool reactor. The method SwitchToConsensus in consensus will have to update the waitSync info in consensus and mempool, creating a new dependency between the two reactors. I think we don't want to do this.

Ideally we should have a separate module that controls the different phases of the node, but that's for another discussion.

Reply:
In the end I changed ConsensusReactor by MempoolReactor. The waitSync information is already in the mempool reactor so I just needed to add MempoolReactor to the environment.